subset/switches/src/main/java/switchtest/SwitchInterrogator.java

```diff
@@ -445,6 +439,49 @@ private String cleanShowRunData(String data) {
     return data;
   }
 
+  private void validateTests() {
+    try {
+      System.out.println("Tests validating....");
+      // switch.port.link test passes is "Link is UP" and dropped packets being equal to 0.
+      if (show_interface_data[1].equals("UP") && Integer.parseInt(show_interface_data[12]) == 0) {
+        login_report += "\nswitch.port.link=true";
+      } else {
+        login_report += "\nswitch.port.link=false";
+      }
+
+      // switch.port.speed test passes is "configured speed to auto" to ensure the switch is
+      // configured to auto negotiate speed and value of auto-negotiated speed "current speed" is
+      // greater or equal than 10 Mbps.
+      if (show_interface_data[4] != null) {
```

Comment:
"show_interface_data" likely needs a better name since it makes no sense in this content.  What really are the contents of that variable?  Changing a variable name should be trivial with Eclipse or Intelli!

Reply:
will solve with a hashmap